src/lib/storage/storage_manager.hpp

```diff
@@ -31,7 +31,7 @@ class StorageManager : public Noncopyable {
   std::shared_ptr<Table> get_table(const std::string& name) const;
   bool has_table(const std::string& name) const;
   std::vector<std::string> table_names() const;
-  const tbb::concurrent_unordered_map<std::string, std::shared_ptr<Table>>& tables() const;
+  std::unordered_map<std::string, std::shared_ptr<Table>> tables() const;
```

Comment:
Can/should we return a `const shared_ptr` so that people don't think they are actually modifying the list of tables?

Reply:
Tidy was complaining about the const here. Since this is not a reference, how should they think they could modify it?

Reply:
If you read the prototype properly, you shouldn't. Did tidy complain about the const inside or about a const on the unordered_map?

Reply:
I didn't understand it either:

```
/var/lib/jenkins/workspace/hyrise_hyrise_PR-2161/src/lib/storage/storage_manager.cpp:81:1: error: return type 'const std::unordered_map<std::string, std::shared_ptr<Table> >' (aka 'const unordered_map<basic_string<char>, shared_ptr<opossum::Table> >') is 'const'-qualified at the top level, which may reduce code readability without improving const correctness [readability-const-return-type,-warnings-as-errors]
const std::unordered_map<std::string, std::shared_ptr<Table>> StorageManager::tables() const {
```

Reply:
Try to move the const in front of the shared_ptr please

Reply:
That does not directly work because the shared_ptr in the actual map is also not const. Making that shared_ptr const as well leads to errors in other places. I need to look at that properly.

Reply:
Thinking about it, I'm not 100% sure if unordered_map supports const values at all.